crates/polars-time/src/windows/duration.rs

```diff
@@ -439,36 +440,84 @@ impl Duration {
         )
     }
 
+    /// Localize result to given time zone., respecting DST fold of original datetime.
+    /// For example, 2022-11-06 01:30:00 CST truncated by 1 hour becomes 2022-11-06 01:00:00 CST,
+    /// whereas 2022-11-06 01:30:00 CDT truncated by 1 hour becomes 2022-11-06 01:00:00 CDT.
+    ///
+    /// * `original_dt_naive` - original datetime, without time zone.
+    ///   E.g. if the original datetime was 2022-11-06 01:30:00 CST, then this would
+    ///   be 2022-11-06 01:30:00.
+    /// * `original_dt_utc` - original datetime converted to UTC. E.g. if the
+    ///   original datetime was 2022-11-06 01:30:00 CST, then this would
+    ///   be 2022-11-06 07:30:00.
+    /// * `result_dt_naive` - result, without time zone.
+    #[cfg(feature = "timezones")]
+    fn localize_result(
+        &self,
+        original_dt_naive: NaiveDateTime,
+        original_dt_utc: NaiveDateTime,
+        result_dt_naive: NaiveDateTime,
+        tz: &Tz,
+        localize_datetime: impl Fn(NaiveDateTime, &Tz, Ambiguous) -> PolarsResult<NaiveDateTime>,
+    ) -> NaiveDateTime {
+        match localize_datetime(result_dt_naive, tz, Ambiguous::Raise) {
```

Comment:
I'd expect it to be quite rare

For example, if you had `s = pl.datetime_range(datetime(2020, 1, 1), datetime(2021, 1, 1), '1m', closed='left', eager=True)`, and truncated by `'1h'`, then it would raise for the 2-hour period during which the datetime is ambiguous.

That would be 60 / 527040 values, i.e. for ~0.01% of the values it would have to do a bit of extra computation

---

> Can we make a localize_datetime that returns an Option<> and a try_localize_datetime (that reuses localize_datetime) and returns an error.

sure, done! haven't re-used `localize_datetime_opt` though, as the error message is different for ambiguous and non-existent datetimes